src/Sema.zig

```diff
@@ -27442,19 +27498,23 @@ fn semaStructFields(mod: *Module, struct_obj: *Module.Struct) CompileError!void
                 break :msg msg;
             };
             return sema.failWithOwnedErrorMsg(msg);
-        } else if (struct_obj.layout == .Packed and !(validatePackedType(field.ty))) {
-            const msg = msg: {
-                const tree = try sema.getAstTree(&block_scope);
-                const fields_src = enumFieldSrcLoc(decl, tree.*, struct_obj.node_offset, i);
-                const msg = try sema.errMsg(&block_scope, fields_src, "packed structs cannot contain fields of type '{}'", .{field.ty.fmt(sema.mod)});
-                errdefer msg.destroy(sema.gpa);
+        } else if (struct_obj.layout == .Packed) {
+            const tree = try sema.getAstTree(&block_scope);
+            const field_src = enumFieldSrcLoc(decl, tree.*, struct_obj.node_offset, i);
```

Comment:
`enumFieldSrcLoc` loads the AST and should avoided in the non-error path. Instead you should pass `.unneeded` and then check if the function returns `error.SourceLocationNeeded` and then call the function again with the proper source location.

Reply:
Good to know, I ended up removing this code as I found some deeper issues with my previous approach.